src/pages/tablePage/componetns/CustomToolbar.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { useState, useEffect, FC } from "react";
+import {
+    GridToolbarContainer,
+    GridToolbarExport,
+    GridToolbarFilterButton,
+    GridToolbarColumnsButton,
+    GridToolbarDensitySelector,
+} from "@mui/x-data-grid";
+import { Button, Menu, MenuItem } from "@mui/material";
+import { UserType } from "types/UserTypes";
+import { useAuthContext } from "context/AuthContext";
+
+interface CustomToolbarProps {
+    selectedUserId: string | null;
+    setSelectedUserId: (userId: string | null) => void;
+}
+
+const CustomToolbar: FC<CustomToolbarProps> = ({ selectedUserId, setSelectedUserId }) => {
+    const [anchorEl, setAnchorEl] = useState<HTMLElement | null>(null);
+    const { clients, fetchClients } = useAuthContext();
+
+    useEffect(() => {
+        const fetchData = async () => {
+            try {
+                await fetchClients();
+            } catch (error) {
+                console.log('Error fetching clients:', error);
+            }
+        };
+
+        fetchData();
+    }, []);
```

Comment:
jak ma Ci sie to odswiezac skoro dependecy array jest pusty? Odpalasz pobieranie klientow tylko raz w momencie zaladowania komponentu i nie dajesz zadnej instrukcji, zeby pobrac znowu po wybraniu userId.